### ISSUE TITLE
Single invalid style filter disables all cosmetic filtering on the page

### ISSUE THREAD

**Issue by Yuki2718** [uBlockOrigin/uBlock-issues#2170]
### Prerequisites

- [X] I verified that this is not a filter issue (MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues))
- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue
- [X] The issue is not present after wholly disabling uBlock Origin ("uBO") in the browser
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### Description

As the title says. See Steps to reproduce.

### A specific URL where the issue occurs

www.backcountry.com

### Steps to Reproduce

1. Add
```
www.backcountry.com##.chakra-portal
www.backcountry.com##body:style(overflow: auto !important;padding: var(--chakra-space-0 !important;)
```
to My filters - note the second rule lacks an close parenthesis for var but neither picker or assets viewer marks as invalid. Also enable EasyList Cookie to see the effect.

2. Visit the page and see newsletter modal and cookie banner are displayed, which should not be the case and the logger shows according filters applied.
3. If you add the close parenthesis (i.e. `www.backcountry.com##body:style(overflow: auto !important;padding: var(--chakra-space-0) !important;)`) everything works as expected<sup/>1</sup>.
___
1. The style filter acutally doesn't work on the page but it's not relevant here.

### Expected behavior

uBO ignores the style filter and other elements are hidden.

### Actual behavior

All cosmetic filters are not applied yet logged.

### uBlock Origin version

1.43.0/1.43.1b3

### Browser name and version

Firefox 102.0.1

### Operating System and version

Windows 10

**Comment by gwarser** [comment]
I'm redirected to `https://www.bergfreunde.eu/?wt_mc=eu.referral.backcountry.backcountry.-&pid=10004&utm_source=backcountry&utm_medium=referral` :/

**Comment by krystian3w** [comment]
Definitely EU locked page.

**Comment by gwarser** [comment]
@gorhill style is leaking into the Dashboard???

![image](https://user-images.githubusercontent.com/886325/179319378-98798fb6-4a75-4557-9366-e03aaa741ddb.png)

**Comment by krystian3w** [comment]
Yes sir.

**Comment by gorhill** [comment]
Oops. I assumed [this line](https://github.com/gorhill/uBlock/commit/a8ca5127c959e133fbba79fa8343d1569960723f#diff-848f3a5c8459fe07d1c65764e30b7c9471be77f9e9574674442319b831138024R1341) would work:

>     styleElement.disabled = true;

**Comment by gorhill** [comment]
How do you reproduce this?

**Comment by gorhill** [comment]
Just found out that cosmetic filters with pseudo elements are deemed invalid in Firefox:

    ##a::before

Is a perfectly valid filter, but it's being rejected in Firefox, while it passes in Chromium.

It seems Firefox's implementation of [CSSRule.selectorText](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleRule/selectorText) is faulty and causing this. There was measurable performance gain in the parser with using CSSRule.selectorText approach, but I will have to fall back to using the generate-a-whole-stylesheet approach.

**Comment by Yuki2718** [comment]
But pseudo elements rules have actually been working on Firefox. For example, on `https://prtimes.jp/story/detail/YxREvyu63OB` if you add
```
prtimes.jp##.detail__content-sns-area:style(position: static!important)
prtimes.jp##.detail__content-sns-items:not(:first-child):not(:last-child)
```
the right sidebar lacks the top edge. Adding `prtimes.jp##.detail__content-sns-area::after` fixes that.

**Comment by krystian3w** [comment]
But this looks like use generic pseudo-elements in picker or very generic like use DandelionSprout: https://github.com/uBlockOrigin/uBlock-issues/issues/464

**Comment by gorhill** [comment]
> But pseudo elements rules have actually been working on Firefox

Yes I know, I meant my latest commit here is causing this undue invalidation.

**Comment by gorhill** [comment]
> use many lines and move line two into line one and delete no neded characters

Need more specific repro steps. What are those lines? Where do you split in two lines? Which characters to delete?

**Comment by krystian3w** [comment]
I tests block pop-up from main plan of z-index layer without use save in my filters tab:

 so I start to hide pop-up, I copy domain form url bar and paste before `##` and duplicate to protect before lost (I don't use often history in clipboard) - fast if I have a few pop-ups with locked scroll in many reported annoying pop-ups (e.g. gdpr so no worth for uBo Annoyances).

Later I test what selector  class is used for locking scroll and e.g. `##.modal-open:style(overflow: visible !important;)` (based on detected `.modal-backdrop` as foreground), I stared delete invalid put domain form url - no match body to unlock scroll (0 match elements) - so I copy both and I opened again picker without refresh page and both filters match if I delete domain and remove paste domain (to faster move filters info file for subscribers).

Bug maybe not related by latest changes in beta I guess (if not to overload some functions), without save log form browser and devtools consoles.
